### PR TITLE
instagram-challenge

### PR DESCRIPTION
**Pull request by blancopado**

### CODE REVIEW

File: spec/features/pictures_feature_spec.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+require_relative 'helpers'
+
+feature 'pictures' do
+	context 'no pictures have been added' do
+		scenario 'should display a prompt to add a picture' do
+			visit '/pictures'
+			expect(page).to have_content 'No pictures yet'
```
Comment (k0zakinio) [inline spec/features/pictures_feature_spec.rb]: split out these expectations

File: spec/controllers/pictures_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe PicturesController, type: :controller do
```
Comment (yevdyko) [inline spec/controllers/pictures_controller_spec.rb]: It's better to delete it.

File: spec/controllers/comments_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe CommentsController, type: :controller do
```
Comment (yevdyko) [inline spec/controllers/comments_controller_spec.rb]: It's better to delete it.

File: app/controllers/pictures_controller.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  def show
+  	@picture = Picture.find(params[:id])
+  end
+
+  def edit
+  	@picture = Picture.find(params[:id])
+  end
+
+  def update
```
Comment (yevdyko) [inline app/controllers/pictures_controller.rb]: If you are redirecting, you do not need to set @ variables

File: app/controllers/pictures_controller.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  	@picture = Picture.find(params[:id])
+    @picture.update(pictures_params)
+    redirect_to '/pictures'
+  end
+
+  def destroy
```
Comment (yevdyko) [inline app/controllers/pictures_controller.rb]: If you are redirecting, you do not need to set @ variables

File: app/models/picture.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class Picture < ActiveRecord::Base
+
```
Comment (yevdyko) [inline app/models/picture.rb]: You should use belongs_to associations here

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (yevdyko) [inline app/models/user.rb]: You should use has_many associations here

File: app/controllers/comments_controller.rb [hunk 8]
```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+
+	before_action :authenticate_user!
+
+	def new
+	  @picture = Picture.find(params[:picture_id])
+	  @comment = Comment.new
+	end
+
+	def create
```
Comment (yevdyko) [inline app/controllers/comments_controller.rb]: If you are redirecting, you do not need to set @ variables

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module ApplicationHelper
```
Comment (yevdyko) [inline app/helpers/application_helper.rb]: Should be deleted if not needed.

File: app/helpers/comments_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (yevdyko) [inline app/helpers/comments_helper.rb]: Should be deleted if not needed.

File: app/helpers/pictures_helper.rb [hunk 11]
```diff
@@ -0,0 +1,2 @@
+module PicturesHelper
```
Comment (yevdyko) [inline app/helpers/pictures_helper.rb]: Should be deleted if not needed.

**Comment by tansaku** [comment]
You had your instagram_challenge reviewed by **Andrew Dowell**.
### The good points are:
* Consistent indentation, Dead code and comments removed, Unnecessary files removed
* Test descriptions consistent, Tests DRY
* Models appropriately tested
* Nested forms used appropriately
* All controller methods short and succinct
* Thin controllers with minimal logic
* Model logic is DRY
* Relative and absolute paths used appropriately
* Semantic HTML elements used appropriately
* Javascript named functions employed appropriately

### You should consider the following improvements:
* Installation/Usage Instructions should be in README
* Ensure Rakefile has Appropriate Tasks
* Gemfile should use Test Groups
* Ensure spec_helper.rb is Configured Correctly
* Ensure Asset Routes are Set Correctly
* Feature tests should be declarative, not imperative
* Delete empty helper files
* No need to set instance vars when redirecting
* Lack of connection between models
* Feature tests should use 'within' for precision
* Tests should separate out expectations where appropriate
* Use Asset Path correctly for Responsive Images
* Use partials appropriately
* Replace instance variables with local variables
* Avoid Law of Demeter Violations

### Additional comments:
Test expectations: Could test for image actually being visible on the page

Model connections: Picture belongs_to :user

No CSS added

No JS used

Good job



see https://github.com/makersacademy/instagram_challenge/blob/master/docs/review.md for more details on the items above

**Pull request closed without merging**